docs/source/notebooks/Introduction to Ablator.ipynb

```diff
@@ -0,0 +1,163 @@
+    "\n",
+    "- Model configuration\n",
+    "\n",
+    "- Training configuration\n",
+    "\n",
+    "- Optimizer and Scheduler configuration\n",
+    "\n",
+    "- Running configurations\n",
+    "\n",
+    "Flexibility is a big plus with this configuration system. It lets researchers tailor their experiments to their specific aims and hypotheses.\n",
+    "\n",
+    "[**Training Module -**](https://ablator-tutorials.readthedocs.io/en/v0.0.1-misc/training.html)\n",
+    "\n",
+    "After setting up the experiment configurations,the trainer module takes these configurations. It handles the execution of the experiment, whether it is a single prototype experiment (using ProtoTrainer) or a parallel experiment with hyperparameter optimization (using ParallelTrainer). This module has a training interface, Model Wrapper, which encapsulates common boilerplate code, abstracts the  repetitive tasks.The Configuration Module and the Trainer Module are blueprints for shaping the experiment.\n",
+    "\n",
+    "[**Experiment result metrics module -**](https://ablator-tutorials.readthedocs.io/en/v0.0.1-misc/results.html)\n",
+    "\n",
+    "This class plays a pivotal role in capturing the performance of models. It receives predictions and outputs after the training of the ML model, and applyies specific evaluation functions to calculate metrics.\n",
+    "\n",
+    "\n",
+    "[**Analysis Module -**](https://ablator-tutorials.readthedocs.io/en/v0.0.1-misc/analysis.html)\n",
+    "\n",
+    "The Analysis module in Ablator provides essential tools for visualizing and interpreting the outcomes. The Analysis module consists of two main classes: `PlotAnalysis` and `Results`. The Analysis module comes into play after the training and evaluation are complete. It takes the experiment results, which include metrics and hyperparameters, and processes them using the `Results` class. The processed data is then visualized using the `PlotAnalysis` class, creating insightful.\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "62252634",
+   "metadata": {},
+   "source": [
+    "\n",
+    "\n",
+    "\n",
+    "\n",
+    "## What happens in the background once ABLATOR is launched? \n",
+    "\n",
+    "Ablator initializes a Ray cluster for parallel execution. During Ablation Studies, Optuna navigates the search space randomly for the trails. In contrast, when conducting Hyperparameter Optimization experiments, Optuna meticulously collects and analyzes their performance to effectively guide the exploration of the search space. \n",
+    "\n",
+    "These trials run in parallel on available resources in the Ray cluster. The experiment directory is synchronized after each trial is completed, regardless of success or failure. This ensures that trial-specific information and results are stored for analysis. \n",
```

Comment:
Maybe add a mention that optuna also uses these results to navigate the search space in an HPO experiment.